submissions/sofiichuk/a-tiny-js-world/index.js

```diff
@@ -0,0 +1,57 @@
+const man = {
+    species: 'human',
+    name: 'oneArmedBandit',
+    gender: 'male',
+    legs: 2,
+    hands: 1,
+    saying: '\"Arrrgh!\"',
+    friends: 0
+};
+
+const woman = {
+    species: 'human',
+    name: 'twoArmedWoman',
+    gender: 'female',
+    legs: 2,
+    hands: 2,
+    saying: '\"How dare you?!\"',
+    friends: 0
+};
+
+const dog = {
+    species: 'dog',
+    name: 'bobik',
+    gender: 'male',
+    legs: 5,
+    hands: 0,
+    saying: '\"I can speak too.What\'s the problem?\"',
+    friends: 'sheMurzik'
+};
+
+const cat = {
+    species: 'cat',
+    name: 'sheMurzik',
+    gender: 'female',
+    legs: 4,
+    hands: 0,
+    saying: '\"I\'m just a normal cat.I meant:\"Meow\"',
+    friends: 'bobik'
+};
+
+const catWoman = {
+    species: 'chimera',
+    name: 'undefined',
+    gender: 'undefined',
+    legs: 'undefined',
+    hands: 'undefined',
+    friends: 'undefined'
+};
+
+Object.setPrototypeOf(catWoman, cat);
```

Comment:
This is the second time I see `Object.setPrototypeOf`, did anyone tell you to use it? [Specs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf) say it is a very slow method and there is an alternative - `Object.create()`. Or you could've used `bind`. Anyway, keep these in mind.

Reply:
at first i tried to transform cat.saying to an object/array to pass it by reference;
then i tried to combine cat and catwoman props using object.assign;
i failed with both;
then i found out people handle this with a __proto__;
but i had already known that using __proto__ in an explicit way is strongly not recommended;
so i continued searchin for smth similar as a substitution;
nth-google-request was "set prototype" and it led to setPrototype method;
(and now i see it's no better than using __proto__ - but already afterwards)

Reply:
Got it, nice research)